File: Data Structures & Algorithms/same-binary-tree/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        pa=deque()
        qa=deque()
        if p and q:
            pass
        elif not p and not q:
            return True
        elif p or q:
            return False
        pa.append(p)
        qa.append(q)
        equiv=True
        while len(pa)>0:
            for i in range(len(pa)):
                pcurr=pa.popleft()
                qcurr=qa.popleft()
                if pcurr and qcurr:
                    if pcurr.val==qcurr.val:
                        pa.append(pcurr.left)
                        pa.append(pcurr.right)
                        qa.append(qcurr.left)
                        qa.append(qcurr.right)
                    else:
                        return False
                elif pcurr or qcurr:
                    return False
                elif not pcurr and not qcurr:
                    pass
        return True

File: Data Structures & Algorithms/invert-a-binary-tree/submission-6.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root==null){
            return null;
        }
        Deque<TreeNode> deque=new ArrayDeque<>();
        deque.offer(root);
        TreeNode temp;
        while(!deque.isEmpty()){
            TreeNode curr=deque.poll();
            temp=curr.left;
            curr.left=curr.right;
            curr.right=temp;
            if (curr.right!=null){
                deque.add(curr.right);
            }
            if (curr.left!=null){
                deque.add(curr.left);
            }


        }
        return root;

    }

}
